pkgs/racket-doc/syntax/scribblings/format.scrbl

```diff
@@ -0,0 +1,106 @@
+              [#:props props (or/c syntax? #f 'infer) 'infer]
+              [#:track track (or/c #t (-> identifier? identifier?) #f) #t]
+              [#:binder? binder? any/c #t])
+         identifier?]{
+
+Builds an @tech[#:doc refman]{identifier} with a name constructed from the @racket[v]s, using
+@racket[context] for the @tech[#:doc refman]{lexical information} and @racket[source] for the source
+location. Each @racket[v] is converted to a symbol made up of the same characters that would be
+produced by @racket[~a], except for @tech[#:doc refman]{identifiers}, which use their symbolic name,
+and @tech[#:doc refman]{keywords}, which are converted using @racket[keyword->string] (i.e. the
+preceeding @litchar{#:} characters are not included).
+
+@(examples
+  #:eval (make-format-eval) #:once
+  (~id #'hyphenated "-" #'id)
+  (~id #'made #\- 'from #\- '#:many #\- "pieces"))
+
+If @racket[props] is a @tech[#:doc refman]{syntax object} or @racket[#f], then
+@tech[#:doc refman #:key "syntax property"]{syntax properties} from @racket[props] are copied to the
+new identifier in the same way as for @racket[datum->syntax]. If @racket[props] is @racket['infer],
+then the properties of the new identifier are copied from @racket[context] if @racket[track] is
+@racket[#f]; if @racket[track] is not @racket[#f], then no properties are copied.
+
+If @racket[track] is not @racket[#f], then a @racket['sub-range-binders] @tech[#:doc refman]{syntax
+property} is added to the new identifier. The property value is a list containing an element for each
+@tech[#:doc refman]{identifier} @racket[_v-id] provided for @racket[v], and each element is an
+immutable @tech[#:doc refman]{vector} with the shape
+
+@(racketblock
+  (vector-immutable _result-id _v-id-position _v-id-length 0.5 0.5
+                    _v-id 0 _v-id-length 0.5 0.5))
+
+where @racket[_result-id] is the new identifier, @racket[_v-id-position] is the position that
+@racket[_v-id]'s name appears in @racket[_result-id], and @racket[_v-id-length] is the length in
+characters of @racket[_v-id]'s symbolic name. This protocol helps Check Syntax understand that uses of
+the identifier should be treated as uses of the @racket[_v-id]s; see
+@seclink["Syntax_Properties_that_Check_Syntax_Looks_For" #:doc '(lib "scribblings/tools/tools.scrbl")
+#:indirect? #t]{Syntax Properties that Check Syntax Looks For} for more details.
+
+@(examples
+  #:eval (make-format-eval) #:once
+  (syntax-property (~id #'foo "-" #'bar) 'sub-range-binders))
+
+If @racket[track] is a procedure, then it is applied to @racket[_result-id] and to each @racket[_v-id]
+inside the @racket['sub-range-binders] property. If @racket[track] is @racket[#t], then
+@racket[syntax-local-introduce] is used if @racket[(syntax-transforming?)] is @racket[#t], otherwise
+@racket[identity] is used. (This default corresponds to the correct behavior when each @racket[_v-id]
+comes from the input to a @tech[#:doc refman]{syntax transformer}.)
```

Comment:
Add a `defproc` for the default?

If you do, then it might be helpful to name the new function as the default for `#:track`, instead of `#t`